### PR TITLE
Issue/7837 synchronize alt field with remote

### LINKED ISSUE #7837: Media: synchronize ALT field with remote.
In order to properly load the ALT field for our media, we need to:

1. Modify `RemoteMedia` to be able to store the `alt` text.
2. Modify `Media` to also be able to store the `alt` text.
3. Modify `MediaServiceRemoteREST.m` and `MediaServiceRemoteXMLRPC.m` to keep the `alt` field synchronized with our remote.
4. Add or modify the appropriate unit tests to make sure our services handle the `alt` field correctly.

### PR DESCRIPTION
**Pull request by 3vangelos**
Fixes #7837

To test:
- alt text which are added through web front end will be saved in the iPhone db (Can be observed with an external CoreData Viewer, like Core Data Editor)
- unit tests added which test parsing of json

Needs review: @diegoreymendez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
@elibud even though the PR is not ready yet, it's been 6 hours since the last commit and BuddyBuild is still blocking this PR.

**Comment by nheagy** [comment]
@3vangelos BuddyBuild missed this PR because it's from your repo, but I just updated its config so it should work. If you get a chance, could you push some minor update to this branch so that BB might pick it up and build it? Not a huge deal, it would just be nice to get a BB build to verify the PR 😄

Thanks! 🙇

File: WordPressKit/WordPressKit/RemoteMedia.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 @property (nonatomic, strong) NSString *title;
 @property (nonatomic, strong) NSString *caption;
 @property (nonatomic, strong) NSString *descriptionText;
+@property (nonatomic, strong) NSString *imageAltTag;
```
[the remark this answers is not in the document]
Comment (3vangelos) [reply WordPressKit/WordPressKit/RemoteMedia.h]: Thank you. Good point. I pushed the changes.

**Comment by elibud** [comment]
@3vangelos please update the file `MIGRATIONS.md` with info about the changes you made to the model, we forgot to mention this. Thank you

**Comment by 3vangelos** [comment]
Hi @elibud, I actually already had realised there is such a file, but at the end forgot to update it. MIGRATIONS.md should be up to date now.

**Comment by elibud** [comment]
:shipit: I don't know what's up with Buddybuild, but it was green before the last commit that only includes changes to `MIGRATIONS.md` so we are good to go.

**Pull request merged**